tyk-headless/tests/deployment-gw-repset_test.yaml

```diff
@@ -0,0 +1,294 @@
+suite: test headless gateway deployment
+templates:
+  - templates/configmap-gateway.yaml
+  - templates/deployment-gw-repset.yaml
+  - templates/secrets.yaml
+  - templates/secret-certs.yaml
+release:
+  name: my-release
+  namespace: my-namespace
+chart:
+  version: 0.12.0
+tests:
+  - it: should match metadata name with default values
+    template: templates/deployment-gw-repset.yaml
+    asserts:
+      - equal:
+          path: apiVersion
+          value:  apps/v1
+      - equal:
+          path: kind
+          value:  DaemonSet
+      - equal:
+          path: metadata.name
+          value:  gateway-my-release-tyk-headless
+      - isSubset:
+          path: metadata.labels
+          content:
+            app: gateway-my-release-tyk-headless
+            chart: tyk-headless-0.12.0
+            release: my-release
+  - it: should match metadata name with overrided name
+    template: templates/deployment-gw-repset.yaml
+    set:
+      fullnameOverride: test-name
+    asserts:
+      - isAPIVersion:
+          of: apps/v1
+      - isKind:
+          of: DaemonSet
+      - equal:
+          path: metadata.name
+          value: gateway-test-name
+      - isSubset:
+          path: metadata.labels
+          content:
+            app: gateway-test-name
+            chart: tyk-headless-0.12.0
+            release: my-release
+  - it: should match spec with default values
+    template: templates/deployment-gw-repset.yaml
+    asserts:
+      - equal:
+          path: spec.minReadySeconds
+          value: 5
+      - isSubset:
+          path: spec.updateStrategy
+          content:
+            type: RollingUpdate
+            rollingUpdate:
+              maxUnavailable: 1
+      - isSubset:
+          path: spec.selector.matchLabels
+          content:
+            app: gateway-my-release-tyk-headless
+            release: my-release
+      - isSubset:
+          path: spec.template.metadata.labels
+          content:
+            app: gateway-my-release-tyk-headless
+            release: my-release
+      - isSubset:
```

Comment:
This is check is duplicated. Already have a similar one above

Reply:
thank you for spotting this! i removed duplicated one in [2f2bb5b](https://github.com/TykTechnologies/tyk-helm-chart/pull/245/commits/2f2bb5b89ce19ef9f7b73719dacf47f35c5824f1)